src/lucky/session_handler.cr

```diff
@@ -4,9 +4,31 @@ class Lucky::SessionHandler
   def call(context : HTTP::Server::Context)
     call_next(context)
 
-    if context.session.changed?
-      context.session.set_session
-      context.cookies.write(context.response.headers)
+    write_session(context)
+    write_cookies(context)
+    check_cookie_size(context)
+  end
+
+  private def write_session(context : HTTP::Server::Context)
+    return if !context.session.changed?
```

Comment:
It’s not documented but so far most of the codebase avoids early returns. In a case like this I think: if context.session.changed? Should work well

Reply:
Nailed it: 8cca80f3